libraries/AdaptiveExpressions/ExpressionFunctions.cs

```diff
@@ -3701,6 +3701,76 @@ private static IEnumerable<object> Flatten(IEnumerable<object> list, int dept)
                     },
                     ReturnType.Number,
                     expr => ValidateArityAndAnyType(expr, 1, 1, ReturnType.String)),
+                new ExpressionEvaluator(
+                    ExpressionType.HasFullDate,
+                    (expr, state, options) =>
+                    {
+                        TimexProperty parsed = null;
+                        bool? value = null;
+                        string error = null;
+                        IReadOnlyList<object> args;
+                        (args, error) = EvaluateChildren(expr, state, options);
+                        if (error == null)
+                        {
+                            if (args[0] is TimexProperty timex)
+                            {
+                                parsed = timex;
+                            }
+                            else if (args[0] is JObject jTimex)
+                            {
+                                parsed = jTimex.ToObject<TimexProperty>();
+                            }
+                            else if (args[0] is string ts)
+                            {
+                                parsed = new TimexProperty(ts);
+                            } 
+                            else
+                            {
+                                error = $"{expr} should be either a TimexProperty or a string";
+                            }
+                        }
+
+                        value = parsed != null && parsed.Year != null && parsed.Month != null && parsed.DayOfMonth != null;
+
+                        return (value, error);
+                    },
+                    ReturnType.Boolean,
+                    expr => ValidateArityAndAnyType(expr, 1, 1, ReturnType.Object)),
```

Comment:
>ValidateArityAndAnyType(expr, 1, 1, ReturnType.Object) [](start = 28, length = 54)

You can validate to object | string, right?  Numbers or arrays are not allowed.

Reply:
Yes, these two functions can have a string or a TimexProperty as an argument. 
Here ReturnType.Object represents a base type which includes string, number and boolean, etc.
The actual type checking is in the body of functions.

Reply:
Our ReturnType.Object today actually works like a `any` not a `object`, so object | number won't be able to reject number or array, so we probably can refine the type checking types later.